Bring FPGA SPI tests to feature-hal-spec-spi

mprse opened the pull request:
### Description

This PR provides the following changes:
- brings SPI master/slave tests designed for the FPGA-Test-Shield to the spi feature branch,
- enables FPGA test shield component for: `K64F, K66F, NUCLEO_F429ZI` (*can FPGA-Test-Shield be connected to these CI boards ?*),
- adds limitation for supported clock modes (K64F),
- adds documentation for FPGA-Test-Shield SPI tests (new HAL API)

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [X] Test update
    [ ] Breaking change

### Reviewers

@jamesbeyond Can you check the test documentation?

mprse committed:
Bring FPGA SPI tests into SPI feature branch

mprse committed:
K64F SPI driver: add limitation for supported clock modes

mprse committed:
Add documentation for FPGA-Test-Shield SPI tests (new HAL API)

jamesbeyond reviewed (changes requested):
The header files for the tests are required

mprse committed:
Add header files for fpga spi tests and map test cases to defined behavior.

Also, update names of the test functions (basic test and fpga tests) so they are more consistent and unique.

jamesbeyond reviewed (approved):
The header files look good

mprse committed:
Disable nanostack-border-router example for NUCLEO_F429ZI

mprse committed:
Restrict tests to Arduino FF only and remove unused symbol

mprse committed:
Disable nanostack-border-router example for K66F

mprse committed:
TARGET_TMPM4G9/spi_api.c: Add missing #endif

mprse committed:
EP_AGORA: Remove SPIF component.

SPI is disabled for this target on spi feature branch.
